CRM/Contribute/BAO/Contribution.php

```diff
@@ -3268,7 +3269,15 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
             self::updateFinancialAccounts($params, 'changeFinancialType');
             /* $params['trxnParams']['to_financial_account_id'] = $trxnParams['to_financial_account_id']; */
             $params['financial_account_id'] = $newFinancialAccount;
-            $params['total_amount'] = $params['trxnParams']['total_amount'] = $params['trxnParams']['net_amount'] = $trxnParams['total_amount'];
+            $changeFTAmount = $trxnParams['total_amount'];
```

Comment:
I think this newly added code can be moved to a new private method called calcluateFTChangeAmount() and rewritten like that , methods should not be big and we should stop adding more lines to them.

```php
list($changeFTAmount, $ignoreChangeAmount) = $this->calcluateFTChangeAmount($params, $trxnParams['total_amount']);


private function calcluateFTChangeAmount($params, $totalAmount) {
    $changeFTAmount = $totalAmount;

    if (isset($params['prevContribution']->total_amount) || isset($params['tax_amount'])) {
      $taxAmount = CRM_Utils_Array::value('tax_amount', $params, 0);
      $changesinTaxAmount = $totalAmount - $params['prevContribution']->total_amount + $params['prevContribution']->tax_amount - $taxAmount;
      if ($changesinTaxAmount == 0) {
        $ignoreChangeAmount = TRUE;
        $changeFTAmount = $totalAmount;
      }
    }

  return array($changeFTAmount, $ignoreChangeAmount);
}
```

also you could probably rewrite it in better way but the key point is to separate the code into more short logical units to improve readability .. also please do not forget to add a doc block for this new private method in case you implemented it  .